### PR TITLE
Add query builder for shadowing options

### PR DESCRIPTION
**Pull request by yux0**
Add query builder for shadowing options

### CODE REVIEW

File: src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    this.stringBuffer = new StringBuffer();
+  }
+
+  private void appendPartialQuery(String query) {
+    if (query == null || query.length() == 0) {
+      return;
+    }
+
+    if (this.stringBuffer.length() != 0) {
+      this.stringBuffer.append(" and ");
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java:114]: nit: use `AND_QUERY`?

File: src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  public QueryBuilder setWorkflowStartTime(TimeFilter timeFilter) {
+    if (timeFilter == null) {
+      return this;
+    }
+
+    Collection<String> timerFilters = Lists.newArrayListWithCapacity(2);
+    if (timeFilter.getMinTimestamp() != null) {
+      timerFilters.add(
+          START_TIME_PLACEHOLDER + " >= " + timeFilter.getMinTimestamp().toEpochSecond());
```
Comment (yycptt) [inline src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java:78]: should be nano second?

File: src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java (hunk 1 above)
Comment (yux0) [reply src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java]: updated

File: src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java (hunk 2 above)
Comment (yux0) [reply src/main/java/com/uber/cadence/internal/shadowing/QueryBuilder.java]: Yes. Updated with a get nano function

**Review by yycptt — APPROVED** [review]

**Pull request merged**